src/App.tsx

```diff
@@ -24,6 +24,11 @@ function App() {
       return;
     }
 
+    if (qrCodeValue.length > 25 && !localStorage.getItem('warnedAboutLongQRCode')) {
+      localStorage.setItem('warnedAboutLongQRCode', 'true');
+      toast.warn(`Yo, Q-Art Codes work better with shorter text. Try a URL shortener, leave off http and www. KISS!`);
+    }
```

Comment:
```suggestion
    if (qrCodeValue.length > 25 && !localStorage.getItem('warnedAboutLongQRCode')) {
      localStorage.setItem('warnedAboutLongQRCode', 'true');
      toast.warn(`Yo, Q-Art Codes work better with shorter text. Try a URL shortener, leave off http and www. KISS!`);
      return;
    }
```

If you don't want the generation to proceed if they're over the limit.

Reply:
rather than a fail this is a warn because it may work, but the probability of a successful scan decreases with increasing code length. there's probably some size where the probability of successful scan is close enough to 0 for it to semantically be a failure state, but that's suggested and loosely enforced by the choice to make it an input rather than a textarea